### PR TITLE
feat: validateFirst support parallel

### LINKED ISSUE #21793: Form.Item 的 validateFirst 属性, 对于自定义的 validator 验证无效!
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-2e6n7?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
在用户名中输入 a

### What is expected?
因为在 Form.Item 上设置了 validateFirst, 应该是第二条 rule 验证失败, 提示错误信息 'username' must be between 5 and 16 characters, 第三和第四条 rule 不执行.

### What is actually happening?
第二条 rule 验证失败, 后续 rule 不执行. 但是在控制台中可以看到第四条 rule 的 ajax 还是发送出去了.

| Environment | Info |
|---|---|
| antd | 4.0.0 |
| React | react |
| System | win10 |
| Browser | Chrome 80.0.3987.122 |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #21793

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Adjust Form.Item `validateFist` to validate in order. Origin logic use `parallel` to validate.      |
| 🇨🇳 Chinese |    调整 Form.Item `validateFirst` 属性为依次执行校验，原并行逻辑改由配置 `parallel` 开启。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/form/index.en-US.md](https://github.com/ant-design/ant-design/blob/form-parallel/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/form-parallel/components/form/index.zh-CN.md)

### CODE REVIEW

File: components/form/index.en-US.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Form field component for data bidirectional binding, validation, layout, and so
 | rules | Rules for field validation. Click [here](#components-form-demo-basic) to see an example | [Rule](#Rule)[] | - |  |
 | shouldUpdate | Custom field update logic. See [below](#shouldUpdate) | boolean \| (prevValue, curValue) => boolean | false |  |
 | trigger | When to collect the value of children node | string | `onChange` |  |
-| validateFirst | Whether stop validate on first rule of error for this field | boolean | false |  |
+| validateFirst | Whether stop validate on first rule of error for this field. Will parallel validate when `parallel` cofigured | boolean \| `parallel` | false | `parallel` 4.5.0 |
```
Comment (xrkffgg) [inline components/form/index.en-US.md:95]:
```suggestion
| validateFirst | Whether stop validate on first rule of error for this field. Will parallel validate when `parallel` cofigured | boolean \| `parallel` | false | `parallel`: 4.5.0 |
```

File: components/form/index.zh-CN.md [hunk 2]
```diff
@@ -93,7 +93,7 @@ const validateMessages = {
 | rules | 校验规则，设置字段的校验逻辑。点击[此处](#components-form-demo-basic)查看示例 | [Rule](#Rule)[] | - |  |
 | shouldUpdate | 自定义字段更新逻辑，说明[见下](#shouldUpdate) | boolean \| (prevValue, curValue) => boolean | false |  |
 | trigger | 设置收集字段值变更的时机 | string | `onChange` |  |
-| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验 | boolean | false |  |
+| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验。设置 `parallel` 时会并行校验 | boolean \| `parallel` | false | `parallel` 4.5.0 |
```
Comment (xrkffgg) [inline components/form/index.zh-CN.md:96]:
```suggestion
| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验。设置 `parallel` 时会并行校验 | boolean \| `parallel` | false | `parallel`: 4.5.0 |
```

**Review by xrkffgg — APPROVED** [review]

File: components/form/index.zh-CN.md [hunk 3]
```diff
@@ -93,7 +93,7 @@ const validateMessages = {
 | rules | 校验规则，设置字段的校验逻辑。点击[此处](#components-form-demo-basic)查看示例 | [Rule](#Rule)[] | - |  |
 | shouldUpdate | 自定义字段更新逻辑，说明[见下](#shouldUpdate) | boolean \| (prevValue, curValue) => boolean | false |  |
 | trigger | 设置收集字段值变更的时机 | string | `onChange` |  |
-| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验 | boolean | false |  |
+| validateFirst | 当某一规则校验不通过时，是否停止剩下的规则的校验。设置 `parallel` 时会并行校验 | boolean \| `parallel` | false | `parallel`: 4.5.0 |
```
Comment (yoyo837) [inline components/form/index.zh-CN.md:96]: 我咋理解这里是应该是串行？

Comment (xrkffgg) [reply components/form/index.zh-CN.md:96]: 没有的。 boolean \| `parallel`   展示的出的效果，表示类型是 boolean 或者 `parallel` ，默认值为 false ，其中 `parallel` 是 4.5.0 版本加的

Comment (afc163) [reply components/form/index.zh-CN.md:96]: markdown table 换行和代码格式是个大问题。

Comment (afc163) [reply components/form/index.zh-CN.md:96]: markdown table 换行和代码格式是个大问题。

Comment (yoyo837) [reply components/form/index.zh-CN.md:96]: 那这算breaking change吧？

Comment (zombieJ) [reply components/form/index.zh-CN.md:96]: 嗯，不过影响不大。设置 `validateFirst` 有一个报错就行了。

**Pull request merged**